src/plugins/console/server/lib/proxy_request.ts

```diff
@@ -57,19 +57,24 @@ export const proxyRequest = ({
     reject = rej;
   });
 
+  const finalUserHeaders = { ...headers };
+  const hasHostHeader = Object.keys(finalUserHeaders).some(key => key.toLowerCase() === 'host');
+  if (!hasHostHeader) {
+    finalUserHeaders.host = hostname;
+  }
+
   const req = client.request({
     method: method.toUpperCase(),
     // We support overriding this on a per request basis to support legacy proxy config. See ./proxy_config.
     rejectUnauthorized: typeof rejectUnauthorized === 'boolean' ? rejectUnauthorized : undefined,
     host: hostname,
-    port: port === '' ? undefined : Number(port),
+    port: port === '' ? undefined : parseInt(port, 10),
     protocol,
     path: `${pathname}${search || ''}`,
     headers: {
-      ...headers,
+      ...finalUserHeaders,
       'content-type': 'application/json',
       'transfer-encoding': 'chunked',
-      host: hostname,
```

Comment:
In the past, I've seen the spread object applied after defining default values, as a way to override the defaults with anything defined in the spread object. Would this work here? If so then we won't need `finalUserHeaders` or `hasHostHeader`.

```js
    headers: {
      'content-type': 'application/json',
      'transfer-encoding': 'chunked',
      host: hostname,
      ...headers,
    },
```

Reply:
That solution would work for some cases but unfortunately the override will only take affect if the header object entry name is the same - it could be `Host` or `HoSt` or etc.

In the latter case (as in the original issue) both `Host` and `host` would be in the headers passed to the endpoint.